src/codeflare_sdk/cluster/cluster.py

```diff
@@ -663,29 +707,48 @@ def _check_aw_exists(name: str, namespace: str) -> bool:
         )
     except Exception as e:  # pragma: no cover
         return _kube_api_error_handling(e, print_error=False)
-
     for aw in aws["items"]:
         if aw["metadata"]["name"] == name:
             return True
     return False
 
 
-# Cant test this until get_current_namespace is fixed
+# Cant test this until get_current_namespace is fixed and placed in this function over using `self`
 def _get_ingress_domain(self):  # pragma: no cover
-    try:
-        config_check()
-        api_client = client.NetworkingV1Api(api_config_handler())
-        if self.config.namespace != None:
-            namespace = self.config.namespace
-        else:
-            namespace = get_current_namespace()
-        ingresses = api_client.list_namespaced_ingress(namespace)
-    except Exception as e:  # pragma: no cover
-        return _kube_api_error_handling(e)
+    config_check()
+
+    if self.config.namespace != None:
+        namespace = self.config.namespace
+    else:
+        namespace = get_current_namespace()
     domain = None
-    for ingress in ingresses.items:
-        if ingress.spec.rules[0].http.paths[0].backend.service.port.number == 10001:
-            domain = ingress.spec.rules[0].host
+
+    if is_openshift_cluster():
+        try:
+            api_instance = client.CustomObjectsApi(api_config_handler())
+
+            routes = api_instance.list_namespaced_custom_object(
```

Comment:
Could this be turned into a get operation?

Reply:
Both comments sound good I can add labels to the ingresses/routes something like `ray-dashboard-name:<dashboard-route/ingress-name>`

Reply:
Right, generally that's a best ractice to scope label names, e.g. `sdk.codeflare.dev/ray-dashboard-name`. Or maybe rely on Kubernetes recommended labels https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/.

Reply:
Awesome, label selector was a suggestion to workaround any generated names, but it's even better without it.

> We can create a follow on issue to convert the openshift_oauth ingress into a route and have the standardised ray-dashboard-{cluster_name} naming scheme so we can use get_namespaced_custom_object.

Absolutely 👍🏼.